### PR TITLE
docs: add note for custom command generator template

### LINKED ISSUE #7405: Bug: Generator Command cannot locate the view template.
### PHP Version

7.4

### CodeIgniter4 Version

4.3.3

### CodeIgniter4 Installation Method

Composer (using `codeigniter4/appstarter`)

### Which operating systems have you tested for this bug?

Windows

### Which server did you use?

apache

### Database

_No response_

### What happened?

I created a generator command like so:
```
php spark make:command TestCommandGenerator --type generator --command make:me-test
```
And updated the file like so:
```php
public function run(array $params)
{
    $this->component = 'Library';
    $this->directory = 'Libraries';
    $this->template = 'me.tpl.php';

    $this->execute($params);
}
```
I have the generator template in the directory: `app/Commands/Generators/Views/me.tpl.php`. But when I try to use the command `php spark make:me-test`, I get the error message:
```
Invalid file: "CodeIgniter\Commands\Generators\Views\me.tpl.php"
```

### Steps to Reproduce

1. Create a new command:  `php spark make:command TestCommandGenerator --type generator --command make:me-test`.
2. Create the generator template and place it within the `app/Commands/Generators/Views` folder.
3. Update the `run()` method of the `TestCommandGenerator.php`  to contain the name of the view template you created above: `$this->template = ''; // name of file here`
5. Run the command `php spark make:me-test`.

### Expected Output

I expect that the view template file would be located within the `app` namespace instead of `CodeIgniter`.

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by sammyskills**
**Description**
Added explanation on how to locate the view template of a custom generator command. Fixes #7405.

**Checklist:**
* [x]  Securely signed commits
* [ ] Component(s) with PHPDoc blocks, only if necessary or adds value
* [ ] Unit testing, with >80% coverage
* [x]  User guide updated
* [ ] Conforms to style guide

### CODE REVIEW

File: user_guide_src/source/cli/cli_generators/001.php [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+namespace Config;
+
+use CodeIgniter\Config\BaseConfig;
+
+class Generators extends BaseConfig
+{
+    public array $views = [
+        // ..
+        'make:awesome-command' => 'App\Commands\Generators\awesomecommand.tpl.php',
```
Comment (datamweb) [inline user_guide_src/source/cli/cli_generators/001.php:11]:
```suggestion
        'make:awesome-command' => 'App\Commands\Generators\Views\awesomecommand.tpl.php',
```

**Review by datamweb — CHANGES_REQUESTED** [review]

File: user_guide_src/source/cli/cli_generators.rst [hunk 2]
```diff
@@ -308,3 +308,14 @@ GeneratorTrait
 
 All generator commands must use the ``GeneratorTrait`` to fully utilize its methods that are used in code
 generation.
+
+*************************************
+Accessing a Custom Generator Template
+*************************************
+
+By default, all generator templates will be looked up at the ``CodeIgniter\Commands\Generators\Views`` namespace.
+To declare a custom location for your custom generator template, you will need to add it to the ``app/Config/Generators.php``
```
Comment (datamweb) [inline user_guide_src/source/cli/cli_generators.rst:317]:
```suggestion
To declare a custom location for your custom generator template, you will need to add it to the **app/Config/Generators.php**
```

File: user_guide_src/source/cli/cli_generators.rst [hunk 3]
```diff
@@ -308,3 +308,14 @@ GeneratorTrait
...
+file. For example, if you have a command ``make:awesome-command`` and your generator template is located within your *app*
+directory ``app/Commands/Generators/Views/awesomecommand.tpl.php``, you would update the config file like so:
```
Comment (datamweb) [inline user_guide_src/source/cli/cli_generators.rst:319]:
```suggestion
directory **app/Commands/Generators/Views/awesomecommand.tpl.php**, you would update the config file like so:
```

**Review by datamweb — COMMENTED** [review]
It does not seem correct to use "`" for these two cases.
As far as I know, "**" is used most of the time.
In any case, I did not see an explanation in [documentation.](https://github.com/codeigniter4/CodeIgniter4/blob/develop/contributing/documentation.rst).

Wait for @kenjis to confirm.

**Comment by sammyskills** [comment]
> It does not seem correct to use "`" for these two cases. As far as I know, "**" is used most of the time. In any case, I did not see an explanation in [documentation.](https://github.com/codeigniter4/CodeIgniter4/blob/develop/contributing/documentation.rst).
> 

> Wait for @kenjis to confirm.

Ok.

**Comment by kenjis** [comment]
As a general rule, we use `**` for in-line file paths, and ` `` ` for source code.

**Review by datamweb — APPROVED** [review]

File: user_guide_src/source/cli/cli_generators.rst [hunk 4]
```diff
@@ -308,3 +308,14 @@ GeneratorTrait
...
 generation.
+
+*************************************
+Accessing a Custom Generator Template
+*************************************
+
+By default, all generator templates will be looked up at the ``CodeIgniter\Commands\Generators\Views`` namespace.
+To declare a custom location for your custom generator template, you will need to add it to the **app/Config/Generators.php**
+file. For example, if you have a command ``make:awesome-command`` and your generator template is located within your *app*
+directory **app/Commands/Generators/Views/awesomecommand.tpl.php**, you would update the config file like so:
```
Comment (paulbalandan) [inline user_guide_src/source/cli/cli_generators.rst:319]: This is wrong. The order of lookup is:
1) The template defined in `app/Config/Generators.php`;
2) The template in the `CodeIgniter\Commands\Generators\Views` namespace.

https://github.com/codeigniter4/CodeIgniter4/blob/848c59860b40355f0de10c1dd9ad3b4761e39b2c/system/CLI/GeneratorTrait.php#L267-L276

**Review by paulbalandan — CHANGES_REQUESTED** [review]

File: user_guide_src/source/cli/cli_generators.rst (hunk 4 above)
Comment (sammyskills) [reply user_guide_src/source/cli/cli_generators.rst]: Thanks for point that out.

File: user_guide_src/source/cli/cli_generators.rst [hunk 5]
```diff
@@ -308,3 +308,16 @@ GeneratorTrait
 
 All generator commands must use the ``GeneratorTrait`` to fully utilize its methods that are used in code
 generation.
+
+*************************************************************
+Declaring the Location of a Custom Generator Command Template
+*************************************************************
+
+The default order of lookup for generator templates is first, the template defined in the **app/Config/Generators.php** file,
+and if not found, the template will be looked up at the ``CodeIgniter\Commands\Generators\Views`` namespace.
```
Comment (paulbalandan) [inline user_guide_src/source/cli/cli_generators.rst:317]:
```suggestion
The default order of lookup for generator templates is (1) the template defined in the **app/Config/Generators.php** file,
and (2) if not found, the template found at the ``CodeIgniter\Commands\Generators\Views`` namespace.
```

**Review by paulbalandan — APPROVED** [review]

**Review by kenjis — APPROVED** [review]

**Comment by kenjis** [comment]
Thank you all!

**Pull request merged**